aten/src/ATen/xpu/profiler/XPUActivityProfiler.cpp

```diff
@@ -0,0 +1,140 @@
+XPUActivityProfilerSession::XPUActivityProfilerSession(
+    XPUActivityApi& xpti,
+    const libkineto::Config& config,
+    const std::set<act_t>& activity_types)
+    : xpti_(xpti), config_(config.clone()), activity_types_(activity_types) {
+  xpti_.setMaxBufferSize(config_->activitiesMaxGpuBufferSize());
+  xpti_.enablePtiActivities(activity_types_);
+}
+
+XPUActivityProfilerSession::~XPUActivityProfilerSession() {
+  xpti_.clearActivities();
+}
+
+// =========== Session Public Methods ============= //
+void XPUActivityProfilerSession::start() {
+  profilerStartTs_ =
+      libkineto::timeSinceEpoch(std::chrono::high_resolution_clock::now());
+}
+
+void XPUActivityProfilerSession::stop() {
+  xpti_.disablePtiActivities(activity_types_);
+  profilerEndTs_ =
+      libkineto::timeSinceEpoch(std::chrono::high_resolution_clock::now());
+}
+
+void XPUActivityProfilerSession::processTrace(logger_t& logger) {
+  traceBuffer_.span = libkineto::TraceSpan(
+      profilerStartTs_, profilerEndTs_, "__xpu_profiler__");
+  traceBuffer_.span.iteration = iterationCount_++;
+  auto gpuBuffer = xpti_.activityBuffers();
+  if (gpuBuffer) {
+    xpti_.processActivities(
+        *gpuBuffer,
+        std::bind(
+            &XPUActivityProfilerSession::handlePtiActivity,
+            this,
+            std::placeholders::_1,
+            &logger));
+  }
+}
+
+void XPUActivityProfilerSession::processTrace(
+    logger_t& logger,
+    libkineto::getLinkedActivityCallback get_linked_activity,
+    int64_t captureWindowStartTime,
+    int64_t captureWindowEndTime) {
+  captureWindowStartTime_ = captureWindowStartTime;
+  captureWindowEndTime_ = captureWindowEndTime;
+  cpuActivity_ = get_linked_activity;
+  processTrace(logger);
+}
+
+std::unique_ptr<libkineto::DeviceInfo> XPUActivityProfilerSession::
+    getDeviceInfo() {
+  return {};
+}
+
+std::vector<libkineto::ResourceInfo> XPUActivityProfilerSession::
+    getResourceInfos() {
+  return {};
+}
+
+std::unique_ptr<libkineto::CpuTraceBuffer> XPUActivityProfilerSession::
+    getTraceBuffer() {
+  return std::make_unique<libkineto::CpuTraceBuffer>(std::move(traceBuffer_));
+}
+
+void XPUActivityProfilerSession::pushCorrelationId(uint64_t id) {
+  xpti_.pushCorrelationID(id, XPUActivityApi::CorrelationFlowType::Default);
+}
+
+void XPUActivityProfilerSession::popCorrelationId() {
+  xpti_.popCorrelationID(XPUActivityApi::CorrelationFlowType::Default);
+}
+
+void XPUActivityProfilerSession::pushUserCorrelationId(uint64_t id) {
+  xpti_.pushCorrelationID(id, XPUActivityApi::CorrelationFlowType::User);
+}
+
+void XPUActivityProfilerSession::popUserCorrelationId() {
+  xpti_.popCorrelationID(XPUActivityApi::CorrelationFlowType::User);
+}
+
+// =========== ActivityProfiler Public Methods ============= //
+const std::set<act_t> kXpuTypes{
+    act_t::GPU_MEMCPY,
+    act_t::GPU_MEMSET,
+    act_t::CONCURRENT_KERNEL,
+    act_t::XPU_RUNTIME,
+    // act_t::EXTERNAL_CORRELATION,
+    // act_t::Overhead,
```

Comment:
We need to support Overhead in next step. Let's talk with PTI team next time.

Reply:
ok

Reply:
We should enable the overhead activity now?

Reply:
ok

Reply:
Here is the legacy code of both kineto(cupti) and xpu profiler. No impact will be caused by this data struct.
We need to add the OVERHEAD activity here to enable the pti overhead:
https://github.com/zejun-chen/pytorch/pull/6/files#diff-11a30835eab7801f23620d212a8b7347c87530432971d8777a92f937d31a0bb9L41-L47.

Long term plan: abstract here to make out-of-tree backend to register their profiler.